### DISCUSSION TITLE
How to get user's email confirmation code

### DISCUSSION THREAD

**Discussion by ashep** [zitadel/zitadel#8072] [Q&A]
Hi there.

I use my own email delivery solution, which is not SMTP-compatible, so I cannot use Zitadel's built-in email-sending features.

When a user is being registered via the [API call](https://zitadel.com/docs/apis/resources/user_service/user-service-add-human-user), I get the email code in response, and I can use it to send an email.

But if the user wants to request the code again, I need to get it somehow from Zitadel.

Is there any API that exists for that? Or the only way to do this is to store the initial code and use it afterward?

Does the code have a lifetime?

Thank you.

**Comment by hifabienne** [comment]
Hi @ashep 
There is a resend code api: https://zitadel.com/docs/apis/resources/user_service/user-service-resend-email-code
The code lifetimes can be configured in the settings secret generator: https://zitadel.com/docs/guides/manage/console/default-settings#secret-generator
